### PR TITLE
Add UserBuilder

### PR DESCRIPTION
**Pull request by yashka713**
`UserBuilder` convert Telegram `user` attributes to a hash, from which a new user will be created in the system.

All new user, which will be created after merging this PR into `master`, will be with all their attributes.

Знімок екрану з 2019-07-17 17-44-20

Migration number 4 makes `users => username` consistent with [Telegram username](https://github.com/atipugin/telegram-bot-ruby/blob/master/lib/telegram/bot/types/user.rb).

### CODE REVIEW

**Comment by yashka713** [comment]
@AZatsepa?

File: db/migrations/4_set_username_field_in_users.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Sequel.migration do
+  change do
+    alter_table(:users) do
+      rename_column :user_name, :username
```
Comment (TheBlackArroVV) [inline db/migrations/4_set_username_field_in_users.rb:4]: Why we want to rename it?

File: db/database.rb [hunk 2]
```diff
@@ -7,23 +7,23 @@ def initialize
...
-    user = users.where(user_name: user_name)
+  def decrease_karma(user_data)
+    user = users.where(username: user_data.username)
 
     if user.first
-      user.update(count: user.first[:count] - 1)
+      update_user_karma_counter(user, user.first[:count] - 1)
     else
-      users.insert(user_name: user_name, count: 0)
+      create_user(user_data)
```
Comment (TheBlackArroVV) [inline db/database.rb]: Do we have default count?

File: db/migrations/4_set_username_field_in_users.rb (hunk 1 above)
Comment (yashka713) [reply db/migrations/4_set_username_field_in_users.rb:4]:
```ruby
module Telegram
  module Bot
    module Types
      class User < Base
        attribute :id, Integer
        attribute :is_bot, Boolean
        attribute :first_name, String
        attribute :last_name, String
        attribute :username, String
        attribute :language_code, String
      end
    end
  end
end
```
This is `User` type, which is defined by `gem` and got from `message.reply_to_message.from`.
In this PR user from our database be consistent with Telegram user.

**Pull request merged**